### ISSUE TITLE
Newer Scipy Version required for vectorized truncnorm

### ISSUE THREAD

**Issue by mlindauer** [automl/SMAC3#111]
Using the vectorized version of the truncnorm (in imputation), we need at least scipy version 0.18..
Using scipy 0.17, I get the following error:

```
>>> truncnorm.stats(a=a, b=b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/fr/fr_fr/fr_tl1023/anaconda3/lib/python3.5/site-packages/scipy/stats/_distn_infrastructure.py", line 921, in stats
    cond = self._argcheck(*args) & (scale > 0) & (loc == loc)
  File "/home/fr/fr_fr/fr_tl1023/anaconda3/lib/python3.5/site-packages/scipy/stats/_continuous_distns.py", line 4220, in _argcheck
    if self.a > 0:
```

Also two unit tests fail with scipy 0.17 -- I'm surprised that our unit test on travis ci do not fail.

If you have not any concerns the new scipy, I will update the requirements.txt and setup.py accordingly.

**Comment by mfeurer** [comment]
I can't tell what version we install on travis-ci (the script only states >= 0.13.1), but I assume it's always the latest available version (0.18.1). Since people are mostly using virtual environments or anaconda, I'm fine with specifying that we need scipy 0.18.1. This incorporates the following changes:

* setup.py
* .travis.yml
* requirements.txt

Also, the numpy version must be increased to at least 1.7.1.

Scipy also uses setuptools now (see [changelog](https://github.com/scipy/scipy/releases)), so we could try to install scipy with the `setup_requires` flag now (I'll open a new issue).

**Comment by mfeurer** [comment]
Working on this in #114

**Comment by KEggensperger** [comment]
Fixed with #114
